### PR TITLE
Added helper classes to create a consumer and consume messages

### PR DESCRIPTION
**Pull request by jhorwit2**

### CODE REVIEW

**Comment by jhorwit2** [comment]
Opened this just for comments. Where would we want to place it? New module?

File: hydra-kafka8/src/main/java/com/addthis/hydra/kafka/consumer/highlevel/StreamConsumer.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        MDC.put("topic", topic);
+
+        this.stream = stream;
+        this.messageHandler = messageHandler;
+        this.messageTimer = messageTimer;
+        this.errorMeter = errorMeter;
+    }
+
+    @Override
+    public void run() {
```
Comment (stevedonnelly) [inline hydra-kafka8/src/main/java/com/addthis/hydra/kafka/consumer/highlevel/StreamConsumer.java]: My personal preference would be to put this loop in an anonymous function in KafkaConsumer.  Fewer classes, and less field/constructor boilerplate.

File: hydra-kafka8/src/main/java/com/addthis/hydra/kafka/consumer/highlevel/KafkaConsumer.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import kafka.javaapi.consumer.ConsumerConnector;
+import kafka.serializer.Decoder;
+
+import static com.addthis.hydra.kafka.consumer.ConsumerUtils.newConsumerConfig;
+import static com.codahale.metrics.MetricRegistry.name;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+
```
Comment (stevedonnelly) [inline hydra-kafka8/src/main/java/com/addthis/hydra/kafka/consumer/highlevel/KafkaConsumer.java]: Would prefer a different name that doesnt overlap with the kafka consumer.  Maybe something like KafkaConsumerExecutor?

**Review by stevedonnelly — COMMENTED** [review]
As far as placement goes, I'm not opposed to putting it in hydra-contrib for now.  A bit surprised this class didn't have an equivalent in the kafka-clients library, but not much point submitting a PR to them since it's kafka 8 specific.
Might be worth moving this and some of the other utilities in hydra-contrib to a new basis-kafka module, but you don't have to do that now if you don't want to.

**Comment by tedpearson** [comment]
If it's kafka 8 specific should we namespace it somehow?

File: hydra-kafka8/src/main/java/com/addthis/hydra/kafka/consumer/highlevel/StreamConsumer.java (hunk 1 above)
Comment (jhorwit2) [reply hydra-kafka8/src/main/java/com/addthis/hydra/kafka/consumer/highlevel/StreamConsumer.java]: I hate having that many indentations. This class is internal to the module and isn't intended to be used by consumers of the module. That would place that loop block 6 levels deep.

**Comment by jhorwit2** [comment]
Although I hate the indentation I made the change. Also renamed to KafkaConsumerService. I don't like putting `Executor` in the name since I'm not trying to implement that.

**Comment by jhorwit2** [comment]
When you say basis-kafka, are you proposing adding this to https://github.com/addthis/basis ?

**Comment by jhorwit2** [comment]
@tedpearson the module this code is under is named `hydra-kafka8`, so I think it's namespaced enough ;).

**Comment by tedpearson** [comment]
Derp yup

**Comment by jhorwit2** [comment]
I like the idea of moving this to a `basis-kafka8` module. Seemed weird to me to include this in `hydra-contrib` when it's not really hyrda specific. I'll move the PR there if there is consensus.

**Comment by stevedonnelly** [comment]
Yes, a new basis-kafka8 module sounds good.  It's ok if you want to have basis-kafka8 depend on hydra-contrib.hydra-kafka8 for now, and one of us can move the rest of the generic kafka utilites from hydra-contrib to basis later.

**Comment by jhorwit2** [comment]
Closing in favor of https://github.com/addthis/basis/pull/13

**Pull request closed without merging**